.github/workflows/release/publish-release.yml

```diff
@@ -0,0 +1,97 @@
+name: "Publis Ice Release"
+
+on:
+  workflow_dispatch:
+    inputs:
+      channel:
+        description: "The release channel (e.g., 3.8, nightly)"
+        required: false
+        default: "nightly"
+      run_id:
+        description: "The GitHub run ID to use for artifacts"
+        required: true
+  workflow_call:
+    inputs:
+      channel:
+        required: true
+        type: string
+      run_id:
+        required: true
+        type: string
+jobs:
+  publish-release:
+    runs-on: 'ubuntu-24.04'
+
+    steps:
```

Comment:
I think these should be separate jobs. In case one fails, they would all run independently.